hathor/indexes/rocksdb_utxo_index.py

```diff
@@ -0,0 +1,412 @@
+        array.extend(struct.pack('>Q', self.amount))
+        assert len(array) == 1 + 32 + 25 + 4 + 8
+
+
+@dataclass(frozen=True)
+class _KeyTimeLock(_SeekKeyTimeLock):
+    tx_id: bytes = b''
+    index: int = -1
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        assert len(self.tx_id) == 32
+        assert self.index >= 0
+
+    def _bytes(self, array: bytearray) -> None:
+        super()._bytes(array)
+        array.extend(self.tx_id)
+        array.append(self.index)
+        assert len(array) == 1 + 32 + 25 + 4 + 8 + 32 + 1
+
+    def to_index_item(self) -> UtxoIndexItem:
+        return UtxoIndexItem(
+            token_uid=self.token_uid,
+            tx_id=self.tx_id,
+            index=self.index,
+            address=self.address,
+            amount=self.amount,
+            timelock=self.timelock,
+            heightlock=None,
+        )
+
+
+@dataclass(frozen=True)
+class _SeekKeyHeightLock(_SeekKeyBase):
+    tag: _Tag = _Tag.HEIGHTLOCK
+    heightlock: int = -1
+    amount: int = -1
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        assert self.tag is _Tag.HEIGHTLOCK
+        assert self.heightlock >= 0
+        assert self.amount > 0
+
+    def _bytes(self, array: bytearray) -> None:
+        super()._bytes(array)
+        array.extend(struct.pack('>I', self.heightlock))
+        array.extend(struct.pack('>Q', self.amount))
+        assert len(array) == 1 + 32 + 25 + 4 + 8
+
+
+@dataclass(frozen=True)
+class _KeyHeightLock(_SeekKeyHeightLock):
+    tx_id: bytes = b''
+    index: int = -1
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        assert len(self.tx_id) == 32
+        assert self.index >= 0
+
+    def _bytes(self, array: bytearray) -> None:
+        super()._bytes(array)
+        array.extend(self.tx_id)
+        array.append(self.index)
+        assert len(array) == 1 + 32 + 25 + 4 + 8 + 32 + 1
+
+    def to_index_item(self) -> UtxoIndexItem:
+        return UtxoIndexItem(
+            token_uid=self.token_uid,
+            tx_id=self.tx_id,
+            index=self.index,
+            address=self.address,
+            amount=self.amount,
+            timelock=None,
+            heightlock=self.heightlock,
+        )
+
+
+def _parse_key(key: bytes) -> _KeyBase:
+    assert len(key) >= 1
+    tag = _Tag(key[0])
+    if tag is _Tag.INVALID:
+        raise ValueError('invalid tag found')
+    elif tag is _Tag.TX_EXISTS:
+        assert len(key) == 1 + 32
+        tx_hash = key[1:]
+        return _KeyTxExists(tag=tag, tx_hash=tx_hash)
+    elif tag is _Tag.NOLOCK:
+        assert len(key) == 1 + 32 + 25 + 8 + 32 + 1
+        return _KeyNoLock(
+            tag=tag,
+            token_uid=_from_internal_token_uid(_InternalUid(key[1:33])),
+            address=get_address_b58_from_bytes(key[33:58]),
+            amount=struct.unpack('>Q', key[58:66])[0],
+            tx_id=key[66:98],
+            index=key[98],
```

Comment:
`index` should be an int in the `_KeyNoLock` class, isn't it?

Reply:
Yes, and it is. This works because in Python a `bytes` instance is basically a `list[int]`, so `key[98]` is an `int`.